processing/src/main/java/org/apache/druid/query/aggregation/first/MultiStringFirstDimensionVectorAggregator.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.druid.query.aggregation.VectorAggregator;
+import org.apache.druid.segment.data.IndexedInts;
+import org.apache.druid.segment.vector.BaseLongVectorValueSelector;
+import org.apache.druid.segment.vector.MultiValueDimensionVectorSelector;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+
+public class MultiStringFirstDimensionVectorAggregator implements VectorAggregator
+{
+  private final BaseLongVectorValueSelector timeSelector;
+  private final MultiValueDimensionVectorSelector valueDimensionVectorSelector;
+  private long firstTime;
+  private final int maxStringBytes;
+  private final boolean useDefault = NullHandling.replaceWithDefault();
+
+  public MultiStringFirstDimensionVectorAggregator(
+      BaseLongVectorValueSelector timeSelector,
+      MultiValueDimensionVectorSelector valueDimensionVectorSelector,
+      int maxStringBytes
+  )
+  {
+    this.timeSelector = timeSelector;
+    this.valueDimensionVectorSelector = valueDimensionVectorSelector;
+    this.maxStringBytes = maxStringBytes;
+    firstTime = Long.MAX_VALUE;
+  }
+
+  @Override
+  public void init(ByteBuffer buf, int position)
+  {
+    buf.putLong(position, Long.MAX_VALUE);
+    buf.put(
+        position + NumericFirstVectorAggregator.NULL_OFFSET,
+        useDefault ? NullHandling.IS_NOT_NULL_BYTE : NullHandling.IS_NULL_BYTE
+    );
+    buf.putLong(position + NumericFirstVectorAggregator.VALUE_OFFSET, 0);
+  }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int position, int startRow, int endRow)
+  {
+    final long[] timeVector = timeSelector.getLongVector();
+    final IndexedInts[] valueVector = valueDimensionVectorSelector.getRowVector();
+    firstTime = buf.getLong(position);
+    int index = startRow;
+    for (int i = startRow; i < endRow; i++) {
+      if (valueVector[i].get(0) != 0) {
+        index = i;
+        break;
+      }
+    }
+
+    final long earliestTime = timeVector[index];
+    if (earliestTime < firstTime) {
+      firstTime = earliestTime;
+      buf.putLong(position, firstTime);
+      buf.put(position + NumericFirstVectorAggregator.NULL_OFFSET, NullHandling.IS_NOT_NULL_BYTE);
+      buf.putInt(position + NumericFirstVectorAggregator.VALUE_OFFSET, valueVector[index].get(0));
+    }
+  }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int numRows, int[] positions, @Nullable int[] rows, int positionOffset)
+  {
+    long[] timeVector = timeSelector.getLongVector();
+    IndexedInts[] values = valueDimensionVectorSelector.getRowVector();
+    for (int i = 0; i < numRows; i++) {
+      int position = positions[i] + positionOffset;
+      int row = rows == null ? i : rows[i];
+      long firstTime = buf.getLong(position);
+      if (timeVector[row] < firstTime) {
+        firstTime = timeVector[row];
+        buf.putLong(position, firstTime);
+        buf.put(position + NumericFirstVectorAggregator.NULL_OFFSET, NullHandling.IS_NOT_NULL_BYTE);
```

Comment:
shouldn't this be checking for the value being null or not? or is the assumption that we never set the null bit here and instead translate it in the get method? If that is the case, why do we need a null byte at all instead of just storing a long and an int in the buffer? Or is it to distinguish the case between 'aggregate' not being called from actually aggregating something? (e.g. an empty group should probably always spit out a null value...)